implementations/elixir/ockam/ockam_hub/test/hub/service/alias_test.exs

```diff
@@ -0,0 +1,65 @@
+defmodule Test.Hub.Service.AliasTest do
+  use ExUnit.Case
+
+  alias Ockam.Hub.Service.Alias, as: AliasService
+  alias Ockam.Hub.Service.Echo, as: EchoService
+  alias Ockam.Router
+  alias Test.Utils
+
+  test "alias test" do
+    {:ok, _worker, worker_address} =
+      Test.Hub.Service.AliasTestWorker.start_link(address: "bdf373h28asdf")
+
+    {:ok, _alias, _alias_address} = AliasService.start_link(address: "cdf373h48asdf")
+    {:ok, _echo, _echo_address} = EchoService.start_link(address: "ddf373h47asdf")
+
+    msg = %{onward_route: [worker_address], return_route: [], payload: Utils.pid_to_string()}
+    Router.route(msg)
+
+    assert_receive(:ok, 5_000)
+  end
+end
+
+defmodule Test.Hub.Service.AliasTestWorker do
+  @moduledoc false
+
+  use Ockam.Worker
+
+  alias Ockam.Message
+  alias Ockam.Router
+  alias Test.Utils
+
+  @echo_address "ddf373h47asdf"
+  @alias_address "cdf373h48asdf"
+
+  @impl true
+  def handle_message(message, state) do
+    case Message.return_route(message) do
+      [] -> registration(message, state)
+      _other -> process(message, state)
+    end
+  end
+
+  defp registration(message, state) do
+    msg = %{
+      onward_route: [@alias_address],
+      return_route: [state.address],
+      payload: %{registration_payload: Message.payload(message), forward_route: @echo_address}
```

Comment:
What's the purpose of the `forward_route` here?

Reply:
According to the docs, `:registration_payload` will be send after registration to `:forward_route`. This behaviour is check out when the status of the alias test is is `:registered`. I have also added a `:messaging` status in order to send message after registration.

Reply:
`registration_payload` and `forward_route` are not in the message payload. They supposed to be in the options then the `Ockam.Hub.Service.Alias.Forwarder` worker is created by the alias service worker. It's an internal API for the alias service.
`registration_payload` is the payload of the registration message, while `forward_route` is the return route of the message. https://github.com/ockam-network/ockam/blob/develop/implementations/elixir/ockam/ockam_hub/lib/hub/service/alias.ex#L27

A worker cannot register a different worker in the alias service ATM, it can only register itself and it receives a registration response message from the forwarder worker. The return route of this response message can be used to send messages to the registered worker through the forwarder.

We can think of the better API for that when we do a proper "registry service", but right now that's how it works.

What do you think would be the best format to document that? Function docs? Process diagram in the moduledoc? Any other ideas on how to make it clear?
Also this test would be one of the reference points to that.

Reply:
Thank you for the clarification. Let me continue with this test.

Although function docs are encouraged by a lot of people, I personally prefer to have as much information as possible in the documentation, so `moduledoc` would be better from my point of view.